Updated parameters check for OpenID

VJalili opened the pull request:
When you have the OpenID URL, you don't need to recreate the provider. Only create the provider, if its not already created.

VJalili committed:
Updated missing parameters check for OpenID

When you have the OpenID URL, you don't need to recreate the provider. Only create the provider, if its not already created.

dannon commented:
@VJalili This could use some refactoring, for sure, thanks for looking at it.  At first glance, it looks like this is set to avoid a validation error -- which may be better handled if we just check for an openid_url first and set that openid_provider variable correctly.  Does that make sense?  I can dig deeper if not.

VJalili commented:
I totally agree on separating the provider check from the URL check. I'll add a URL check to better handle validation.

VJalili committed:
Inform that the default provider is being used.

VJalili committed:
Updated all `openid_url` and `openid_provider` checks

VJalili committed:
Corrected Yahoo OpenID endpoint

At the time of making this change, Yahoo OpenID endpoint has issues--It's not accessible. 
This should be checked again later.

VJalili commented:
@dannon I updated all `openid_url` and `openid_provider` checks. All are simplified to 3 checks. 

I think at [this](https://github.com/galaxyproject/galaxy/pull/4039/files#diff-4ee48e7674940f5ea4a6486f7d680e7cL95) point, we would better return with something like: 
`return trans.show_error_message( 'An OpenID provider was not specified' )`

However, Since the previous flow would only update the `message` if neither a provider nor a provider URL is specific, then I followed the same logic in this commit. But maybe it would be better if we just return an error message saying that "a provider is not specified". 

Another point: I think we should remove OpenID URL textbox and all the related parameters/arguments. Because (1) each of the providers we support has a URL associated with, so we don't need a URL provided by the user, (2) I think its a strong expectation that a user should know what the "OpenID" is, search for its discovery endpoint, and enter the endpoints in the URL textbox.

jmchilton commented:
> I think we should remove OpenID URL textbox and all the related parameters/arguments. Because (1) each of the providers we support has a URL associated with, so we don't need a URL provided by the user, (2) I think its a strong expectation that a user should know what the "OpenID" is, search for it's discovery endpoint, and enter the endpoints in the URL textbox.

Agreed - that would be awesome. I've had a 👍 on this for a while so I'm going to go ahead and merge. Sorry for the delay.

VJalili commented:
@jmchilton Thanks. 

If interested, you may have a look at [this](https://github.com/VJalili/galaxy/tree/oauth2) branch (it will soon be ready for a PR). This branch leverages on OAuth2.0 and OpenID Connect protocols (current trending authentication/authorization protocols) for user authentication/authorization flow.

nsoranzo commented on lib/galaxy/webapps/galaxy/controllers/user.py:
```diff
@@ -78,24 +78,21 @@ def openid_auth( self, trans, **kwd ):
+        redirect = kwd.get('redirect', '').strip()
+        if not redirect:
+            redirect = ' '
+        if openid_provider_obj:
```
Sorry I'm late on this!
I think that if both `openid_url` and `openid_provider` are empty, `openid_provider_obj` would not be defined, resulting in a `NameError` exception at this line.
